subsys/usb/class/netusb/function_rndis.c

```diff
@@ -0,0 +1,1000 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_USB_DEVICE_LEVEL
+#define SYS_LOG_DOMAIN "function/rndis"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+
+#include <usb_device.h>
+#include <usb_common.h>
+
+#include <net/ethernet.h>
+#include <net/net_pkt.h>
+#include <net_private.h>
+
+#include <class/usb_cdc.h>
+
+#include "netusb.h"
+#include "function_rndis.h"
+
+/* RNDIS handling */
+#define CONFIG_RNDIS_TX_BUF_COUNT	5
+#define CONFIG_RNDIS_TX_BUF_SIZE	512
+NET_BUF_POOL_DEFINE(rndis_tx_pool, CONFIG_RNDIS_TX_BUF_COUNT,
+		    CONFIG_RNDIS_TX_BUF_SIZE, 0, NULL);
+static struct k_fifo rndis_tx_queue;
+
+/* Serialize RNDIS command queue for later processing */
+#define CONFIG_RNDIS_CMD_BUF_COUNT	2
+#define CONFIG_RNDIS_CMD_BUF_SIZE	512
+NET_BUF_POOL_DEFINE(rndis_cmd_pool, CONFIG_RNDIS_CMD_BUF_COUNT,
+		    CONFIG_RNDIS_CMD_BUF_SIZE, 0, NULL);
+static struct k_fifo rndis_cmd_queue;
+
+struct k_delayed_work notify_work;
+
+/*
+ * Stack for cmd thread
+ */
+static K_THREAD_STACK_DEFINE(cmd_stack, 1024);
+static struct k_thread cmd_thread_data;
+
+static struct __rndis {
```

Comment:
This struct can be better packed by reorganizing its members so that there are no holes due to alignment.

Reply:
I am not sure we need to organize members of the structures by alignment rather then by logical order.

Reply:
I agree with @lpereira here, we should align structs for minimizing memory consumption. The ordering of items might look funny but that is not very relevant. You could add more comments into this struct to explain more what the various integers like net_filter, ep_in, ep_notify are in this struct.

Reply:
Actually I will remove most of fields since we have now new interface for netusb.